### PR TITLE
gh-93464: [Enum] Add versionchanged tag

### PR DESCRIPTION
**Pull request by ethanfurman**
* Issue: gh-93464

### CODE REVIEW

File: Doc/library/enum.rst [hunk 1]
```diff
@@ -804,6 +804,11 @@ Utilities and Decorators
       * ``THREE = [auto(), -3]`` will *not* work (``<auto instance>, -3`` is used to
         create the ``THREE`` enum member)
 
+   .. versionchanged:: 3.11.1
+
+      Prior to 3.11.1 ``auto()`` had to be the only thing
+      on the assignment line to work properly.
```
Comment (CAM-Gerlach) [inline Doc/library/enum.rst]:
```suggestion
      In prior versions, ``auto()`` had to be the only thing
      on the assignment line to work properly.
```

Add missing comma and avoid repeating the version.

**Review by CAM-Gerlach — COMMENTED** [review]
One textual suggestion, otherwise LGTM, thanks!

**Comment by ethanfurman** [comment]
Thanks, that's much better.

**Review by CAM-Gerlach — APPROVED** [review]
LGTM, thanks!

**Review by hugovk — APPROVED** [review]

**Pull request merged**